presidio-anonymizer/tests/test_anonymizer_engine.py

```diff
@@ -255,3 +256,61 @@ def _operate(
     return EngineResult(
         "Number: I am your new text!", [OperatorResult(0, 35, "type", "text", "hash")]
     )
+
+
+def test_when_text_entities_conflict_selected_then_all_conflicts_handled():
+    engine = AnonymizerEngine()
+    text = (
+        "Fake card number 4151 3217 6243 3448.com that "
+        "overlaps with nonexisting URL."
+    )
+    analyzer_result1 = RecognizerResult("CREDIT_CARD", 17, 36, 0.8)
+    analyzer_result2 = RecognizerResult("URL", 32, 40, 0.8)
+    operator_config = OperatorConfig("keep")
+    conflict_strategy = ConflictResolutionStrategy.TEXT_AND_ENTITIES
+    result = engine.anonymize(
+        text,
+        [analyzer_result1, analyzer_result2],
+        {"DEFAULT": operator_config},
+        conflict_resolution=conflict_strategy
+    ).text
+
+    assert result == text
```

Comment:
Should we put the score into consideration as well? Right now the logic would reduce the span of the later entity, regardless of score. Just thinking, maybe we should decide what to trim based on the score. WDYT?

Reply:
Yes right so updated the logic by considering the score as well while changing the entity spans. Now it will follow below cases while comparing between two entities.
1) If score are same or first entity score is higher than the other then it will give priority to first one such that it will adjust the spans of second entity only.
2) If second entity score is higher than first one, then it will give priority to second one and adjust the spans of first one only.

Reply:
Thanks!